package.json

```diff
@@ -0,0 +1,27 @@
+{
+  "name": "in-toto-layout-tool",
+  "version": "1.0.0-beta.0",
+  "description": "A Flask-based web app that helps to create a custom in-toto software supply chain layout",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/in-toto/layout-web-tool.git"
```

Comment:
Not sure, will look into it.

Reply:
Why again would you want to pin a sha1 on this thing?

Reply:
we have a whole [paper explaining](https://www.usenix.org/conference/usenixsecurity16/technical-sessions/presentation/torres-arias) how using weak refs may introduce vulnerabilities. I'd err on the side of pinning sha1's everywhere until better security mechanisms (e.g., in-toto :P) are in place.

Reply:
Yes sure, but I don't think that these vulnerabilities are relevant in this case.
From the [package.json docs](https://docs.npmjs.com/files/package.json#repository):

>Specify the place where your code lives. This is helpful for people who want to contribute.
> [...]
>The URL should be a publicly available (perhaps read-only) url that can be handed directly to a VCS program without any modification.

So no sha1 pinning.
Maybe I should strip down the entire `package.json` to use only the fields that are required for our purpose, i.e. manage front-end dependencies. What do you think?

Reply:
> Yes sure, but I don't think that these vulnerabilities are relevant in this case.

Can you elaborate?

> So no sha1 pinning.

I'm almost sure you can use the `#ref` construction, as this just likely passes this on to git or curl. This makes sense to me because you quote the following yourself:

>The URL should be a publicly available (perhaps read-only) url that can be handed directly to a VCS program without any modification.

The *VCS program without any modification* in this case is git.

To wrap up, the url that you're using uses a weak ref (the default branch, master), that's passed *as is* to git/curl/libgit and likely cloned under some circumstances. From your own source: 

>  It should not be a url to an html project page that you put in your browser. It's for computers.

Reply:
Let's not waste too much time with this: 6f8e13d7f4f38c81d04999c9f6cc097315370bb9